src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -665,84 +668,76 @@ async function fetchAndUseVersions() {
-  const secondaryClickTransmitter = document.querySelector(".secondary-toggle");
+
+  // These are the hamburger-style buttons in the header nav bar. They only
+  // appear at narrow screen width.
+  const primaryToggle = document.querySelector(".primary-toggle");
+  const secondaryToggle = document.querySelector(".secondary-toggle");
+
+  // Cut nodes and classes from `from`, paste into/onto `to`
+  const cutAndPasteNodesAndClasses = (from, to) => {
+    Array.from(from.childNodes).forEach((node) => to.appendChild(node));
+    Array.from(from.classList).forEach((cls) => {
+      from.classList.remove(cls);
+      to.classList.add(cls);
+    });
+  };
+
+  // Hook up the ways to open and close the dialog
   [
-    [primaryClickTransmitter, primaryToggle, primarySidebar],
-    [secondaryClickTransmitter, secondaryToggle, secondarySidebar],
-  ].forEach(([clickTransmitter, toggle, sidebar]) => {
-    if (!clickTransmitter) {
+    [primaryToggle, primaryDialog, primarySidebar],
+    [secondaryToggle, secondaryDialog, secondarySidebar],
+  ].forEach(([toggleButton, dialog, sidebar]) => {
+    if (!toggleButton || !dialog || !sidebar) {
       return;
     }
-    clickTransmitter.addEventListener("click", (event) => {
+
+    // Clicking the button can only open the sidebar, not close it.
+    // Clicking the button is also the *only* way to open the sidebar.
+    toggleButton.addEventListener("click", (event) => {
       event.preventDefault();
       event.stopPropagation();
-      toggle.checked = !toggle.checked;
-
-      // If we are opening the sidebar, move focus to the first focusable item
-      // in the sidebar
-      if (toggle.checked) {
-        // Note: this selector is not exhaustive, and we may need to update it
-        // in the future
-        const tabStop = sidebar.querySelector("a, button");
-        // use setTimeout because you cannot move focus synchronously during a
-        // click in the handler for the click event
-        setTimeout(() => tabStop.focus(), 100);
-      }
+
+      // When we open the dialog, we cut and paste the nodes and classes from
+      // the widescreen sidebar into the dialog
+      cutAndPasteNodesAndClasses(sidebar, dialog);
+
+      dialog.showModal();
     });
-  });
 
-  // Escape key -
-  //
-  // When sidebar is open, user should be able to press escape key to close the
-  // sidebar.
-  [
-    [primarySidebar, primaryToggle, primaryClickTransmitter],
-    [secondarySidebar, secondaryToggle, secondaryClickTransmitter],
-  ].forEach(([sidebar, toggle, transmitter]) => {
-    if (!sidebar) {
-      return;
-    }
-    sidebar.addEventListener("keydown", (event) => {
+    // Listen for clicks on the backdrop in order to close the dialog
+    dialog.addEventListener("click", closeDialogOnBackdropClick);
+
+    // We have to manually attach the escape key because there's some code in
+    // Sphinx's search-highlight.js that prevents the default behavior of the
```

Comment:
Need to file bug report for this. I couldn't find any existing bug report.

Reply:
Did you fill a bug report ? Want to reference it here ?

Reply:
I'm working on it